engine/table/src/main/java/io/deephaven/engine/util/LayoutHintBuilder.java

```diff
@@ -108,9 +111,60 @@ static AutoFilterData fromString(String string) {
         }
     }
 
+    private static class ColumnGroup {
+
+        private final String name;
+        private final List<String> children;
+        private final Color color;
+
+        public ColumnGroup(String name, List<String> children, Color color) {
+            NameValidator.validateColumnName(name);
+            children.forEach(c -> NameValidator.validateColumnName(c));
+
+            this.name = name;
+            this.children = children;
+            this.color = color;
+        }
+
+        @NotNull
+        public String serialize() {
+            StringBuilder sb = new StringBuilder("name:" + name);
+
+            sb.append("::children:" + StringUtils.joinStrings(children, ","));
+            if (color != null) {
+                sb.append("::color:" + "#" + Integer.toHexString(color.javaColor().getRGB()).substring(2));
+            }
+            return sb.toString();
```

Comment:
Sorry, I should have called this out explicitly, but it is the same problem as below. This isn't a fatal issue, but it is "wrong" and to be avoided.

As with the code you changed below, here we have a collection of strings that need to be concatenated together, along with extra contents before and after. 

This is pretty minor in the grand scheme of things, but mixing building big strings in a helper method, then concating that long string with a small string, then appending that entire string "efficiently" into a string builder... If you're aiming for readability, just do away with the StringBuilder entirely and stick +s everywhere, but if you want the extra efficiency of the StringBuilder to handle that conditional below, try to avoid building little strings into big strings, and then big strings into bigger strings, but just accumulate the little strings into the bigger strings.

In JS, strings are usually implemented as "ropes", which can be thought of very loosely as a linked list or tree of actual character arrays, so that insertion at the beginning or end is cheap, and doesn't reallocate everything - but you, the developer, have very little control over what the engine will do when you actually want to say "ok I'm done building, make it all one byte array now".

In contrast, the Java string implementation is pretty nearly just an immutable wrapper over byte[] or char[], where you're expected to ... pretty much avoid expensive things, or they are expensive. A handful of cases are handled automatically and cheaply, but they basically boil down to "one long expression of string concatenation will be replaced by a StringBuilder with its append calls". Downside, you have to work a bit to get performance, upside, optimal performance is actually optimal.

Notes, sort of chronologically with this code, and a proposed set of changes to understand:
 * Don't `+` within append()s (and the constructor is sort of the first append). You can chain append calls if it improves readability, but split all `+`s up with appends instead.
 * Nested method calls that do lots of sub-stringbuilder or sub-`+` work is also not good
```suggestion
            StringBuilder sb = new StringBuilder("name:").append(name);

            sb.append("::children:");
            boolean first = false;
            for (String child : children) {
                if (!first) {
                    sb.append(",");
                }
                first = false;
                sb.append(child);
            }
            if (color != null) {
                sb.append("::color:#")
                    .append(Integer.toHexString(color.javaColor().getRGB()).substring(2));
            }
            return sb.toString();
```